File: 2025/08 August/10 Palindrome SubStrings/main.cpp
// Link: https://www.geeksforgeeks.org/problems/count-palindrome-sub-strings-of-a-string0652/1

#include <iostream>
using namespace std;

class Solution
{
public:
    int countPS(string &s)
    {
        int n = s.size();
        int count = 0;

        for (int i = 0; i < s.size(); i++)
        {
            int left = i - 1;
            int right = i + 1;
            while (left >= 0 and right < n)
            {
                if (s[left] == s[right])
                    count++;
                else
                    break;
                left--;
                right++;
            }
        }

        for (int i = 0; i < s.size(); i++)
        {
            int left = i;
            int right = i + 1;
            while (left >= 0 and right < n)
            {
                if (s[left] == s[right])
                    count++;
                else
                    break;
                left--;
                right++;
            }
        }
        return count;
    }
};

int main()
{
    Solution sol;
    string s = "abaab";
    int result = sol.countPS(s);
    cout << result << endl;

    s = "aaa";
    result = sol.countPS(s);
    cout << result << endl;

    s = "abbaeae";
    result = sol.countPS(s);
    cout << result << endl;

    return 0;
}

File: 2025/08 August/10 Palindrome SubStrings/main.py
# Link: https://www.geeksforgeeks.org/problems/count-palindrome-sub-strings-of-a-string0652/1


class Solution:
    def countPS(self, s):
        n = len(s)
        count = 0

        for i in range(len(s)):
            left = i - 1
            right = i + 1
            while left >= 0 and right < n:
                if s[left] == s[right]:
                    count += 1
                else:
                    break
                left -= 1
                right += 1

        for i in range(len(s)):
            left = i
            right = i + 1
            while left >= 0 and right < n:
                if s[left] == s[right]:
                    count += 1
                else:
                    break
                left -= 1
                right += 1

        return count


if __name__ == "__main__":
    sol = Solution()
    s = "abaab"
    result = sol.countPS(s)
    print(result)

    s = "aaa"
    result = sol.countPS(s)
    print(result)

    s = "abbaeae"
    result = sol.countPS(s)
    print(result)
